libs/s25main/Cheats.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Cheats.h"
+#include "CheatCommandTracker.h"
+#include "network/GameClient.h"
+#include "world/GameWorldBase.h"
+
+Cheats::Cheats(GameWorldBase& world) : cheatCmdTracker_(std::make_unique<CheatCommandTracker>(*this)), world_(world) {}
+
+Cheats::~Cheats() = default;
+
+void Cheats::trackKeyEvent(const KeyEvent& ke)
+{
+    if(!canCheatModeBeOn())
+        return;
+
+    cheatCmdTracker_->trackKeyEvent(ke);
+}
+
+void Cheats::trackChatCommand(const std::string& cmd)
+{
+    if(!canCheatModeBeOn())
+        return;
+
+    cheatCmdTracker_->trackChatCommand(cmd);
+}
+
+void Cheats::toggleCheatMode()
+{
+    if(!canCheatModeBeOn())
+        return;
+
+    isCheatModeOn_ = !isCheatModeOn_;
```

Comment:
Early returns make it a bit harder to follow the logic, especially here I think this is more readable (similar in the other functions):

```suggestion
    if(!canCheatModeBeOn())
       isCheatModeOn_ = !isCheatModeOn_;
```

Reply:
I'll just say "agree to disagree" and replace the early returns. But a time will come when there are enough conditions that I will put my stake in the ground :D